src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/MappedMemorySegmentImpl.java

```diff
@@ -116,13 +116,17 @@ public static MemorySegment makeMappedSegment(Path path, long bytesOffset, long
         if (bytesOffset < 0) throw new IllegalArgumentException("Requested bytes offset must be >= 0.");
         try (FileChannelImpl channelImpl = (FileChannelImpl)FileChannel.open(path, openOptions(mapMode))) {
             UnmapperProxy unmapperProxy = channelImpl.mapInternal(mapMode, bytesOffset, bytesSize);
-            MemoryScope scope = MemoryScope.createConfined(null, unmapperProxy::unmap, null);
-            int modes = defaultAccessModes(bytesSize);
-            if (mapMode == FileChannel.MapMode.READ_ONLY) {
-                modes &= ~WRITE;
+            if (unmapperProxy != null) {
+                MemoryScope scope = MemoryScope.createConfined(null, unmapperProxy::unmap, null);
+                int modes = defaultAccessModes(bytesSize);
+                if (mapMode == FileChannel.MapMode.READ_ONLY) {
+                    modes &= ~WRITE;
+                }
+                return new MappedMemorySegmentImpl(unmapperProxy.address(), unmapperProxy, bytesSize,
+                        modes, scope);
+            } else {
+                return new EmptyMappedMemorySegmentImpl();
```

Comment:
While a little pedantic, should the `mapMode` be considered so that an empty mapped segment will feature the expected access modes when queried through the `accessModes` API ?

Reply:
Ugh - yes, good point